### DISCUSSION TITLE
ScrollArea vertical reversed

### DISCUSSION THREAD

**Discussion by Miloshinjo** [radix-ui/primitives#1449] [Help]
Hello,

I have a task where I need to make scroll area scroll from bottom -> top. Easiest way to do this is via `column-reverse` from flexbox, but UI looks fine, but thumb is not moving when I'm scrolling. Is there something I'm doing wrong, and also, does ScrollArea support that kind of behavior? 

Example: Imaging scrolling slack chat from bottom to top. That's the behaviour I'm trying to implement with ScrollArea.

Thanks

**Comment by benoitgrelard** [comment]
Hey @Miloshinjo, I'm guessing it was you who asked on twitter also, I replied there: https://twitter.com/benoitgrelard/status/1533783056707878913

> Hey, it is just a regular scrolling container underneath so anything you can do with a regular one should be possible here. What is the use case? Like a chat app where it auto-scrolls down all the time?

It looks like I was spot-on with what your use-case was! haha

> Easiest way to do this is via column-reverse from flexbox, but UI looks fine, but thumb is not moving when I'm scrolling. Is there something I'm doing wrong, and also, does ScrollArea support that kind of behavior?

Regarding this, could you provide a sandbox so I could see the issue?

> Example: Imaging scrolling slack chat from bottom to top. That's the behaviour I'm trying to implement with ScrollArea.

Typically the way this would be implemented wouldn't be to reverse the order of the elements, but rather to control the scroll position. You'd run an effect to scroll to the bottom automatically anytime a new message is added.

**Reply by Miloshinjo** [reply]
Thank you for your reply! :) 

I was thinking about controling the scroll position as well, but I have some pagination and stuff running based on what is in view, I will try with useLayoutEffect. But it's interesting that ScrollArea only supports scrolling from top down. 

I also can't get elements that are position absolute to scroll. 

I'm trying to implement it into a big project so I can control scrolls, but many different use cases and it doesn't work for all. :) sad for me, cause I love radix.

If I am able to create code sandbox, I will, but project is too big and I can't share any code due to legal reasons :D

**Reply by benoitgrelard** [reply]
Yeah if you could create smaller isolated sandbox, it would help us understand what you're trying to build and I'm sure we can help.

**Reply by Miloshinjo** [reply]
I will give it a shot later, I'm sorry I can't do it right away :)

**Reply by Miloshinjo** [reply]
Here is my code sandbox reproduction based on your example from the website. Let me know if I can help with code. 

https://codesandbox.io/s/silent-glade-beppek?file=/App.js

 I use `column-reverse` on the content to actually reverse it, and on the `ScrollArea.Viewport` to scroll reverse and on the `ScrollArea.Scrollbar` so scroll bar starts from the bottom. 

You can see thumb is not moving.

This is preferred way for me because I don't have shifts and jumps after `useEffect` or `useLayoutEffect` tho. I do have a hook that controls scroll, but it serves the purpose to remember the scroll position when user returns to this. 

If I go the `useEffect` route, I will have to calculate the max scroll of a div and scroll manually.

**Reply by benoitgrelard** [reply]
Yeah that sandbox does look like a bug, I've opened an issue from this discussion.

**Reply by Miloshinjo** [reply]
Thanks, looks like something that would be nice to be supported for reverse scrolling UIs. I ended up having to scrap the POC because I could not implement it on our project :(

**Reply by benoitgrelard** [reply]
Yeah I agree. Our aim with `ScrollArea` is for it to behave just like a regular native scrolling container, we just hook onto native events and replace the UI. It looks like we potentially have a bug with how we've handled the delta values when it's reversed 👍

**Reply by 0arm** [reply]
Any updates on this?

**Reply by benoitgrelard** [reply]
The related issue is here: #1585.
No updates, I haven't had a chance to look at it yet.

**Reply by Anu-cool-007** [reply]
I'm in a similar boat with a chat screen. Hoping this gets fixed soon

**Reply by maximececcaldi** [reply]
Hello, any updates now?
Thanks

**Reply by blixt** [reply]
I also ran into an issue here. A problem is that ScrollArea injects a `<div style="display:table;…">` element as a child between the viewport element and the actual children, plus when used with shadcn, the forwarded ref goes to the outer element, not the viewport. So you have a bunch of problems implementing a chat UI:

- You can't easily access the scrollable viewport element to check its offset (this can be solved by not using shadcn)
- It's hard to grow the inner children to fill the vertical space, so that you can have bottom-aligned elements in the scroll area, because there is no flex box in the actual scrollable content

I need to experiment more, maybe asChild can take over this strange `display:table` element, but this component is not friendly for the chat use case either way.

**Reply by isaachinman** [reply]
@blixt Did you find a solution? Having to bail out and use native browser scroll behaviour is a failure, as that's the entire problem that `ScrollArea` is supposed to solve...

**Comment by marnec** [comment]
I replied with a solution for my usecase in similar issue #1585

**Comment by mttetc** [comment]
Still no real solution to this?
And no, running an effect to set the scroll position is not one, it's inefficient and introduces tons of things to handle
